### PR TITLE
feat: Introduce - Add validation to the service for incoming messages

### PR DESCRIPTION
**Pull request by soluchok**
* added ProblemReport structure (a few fields)
* now, we can handle one more use case (no one responded with invitation)
* tests were refactored

Part one of issue #890 

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/common/model/problem_report.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package model
+
+// ProblemReport problem report definition
+// TODO: need to provide structure like it is described here
```
Comment (troyronda) [inline pkg/didcomm/common/model/problem_report.go]: issue number?

Comment (soluchok) [reply pkg/didcomm/common/model/problem_report.go]: Done! https://github.com/hyperledger/aries-framework-go/issues/912

File: pkg/didcomm/protocol/introduce/service_test.go [hunk 2]
```diff
@@ -1423,42 +1020,365 @@ func TestService_ProposalUnusualWithRequest(t *testing.T) {
 
 	wg.Add(3)
 
-	go handleIntroducer(&flow{
+	go checkAndHandle(&flow{
```
Comment (troyronda) [inline pkg/didcomm/protocol/introduce/service_test.go]: it would be good to comment each usage of checkAndHandle, since it is now a generic func.

Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service_test.go]: Done!

**Pull request merged**